<!DOCTYPE html>
<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      Quality control & Adapter trimming (II) &middot; Training Course on Next generation sequencing
    
  </title>

  <!-- CSS -->
    <!-- Latest compiled and minified CSS -->
  <link rel="stylesheet" href="/NGSBASE//public/css/poole.css">
  <link rel="stylesheet" href="/NGSBASE//public/css/syntax.css">
  <link rel="stylesheet" href="/NGSBASE//public/css/lanyon.css">
  <link rel="stylesheet" href="/NGSBASE//public/css/styles.css">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400">

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/NGSBASE//public/apple-touch-icon-precomposed.png">
  <link rel="shortcut icon" href="/NGSBASE//public/favicon.ico">

  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
  <script src="/NGSBASE//js/toc.js"></script>
</head>


  <body class="theme-base-07">

    <!-- Target for toggling the sidebar `.sidebar-checkbox` is for regular
     styles, `#sidebar-checkbox` for behavior. -->
<input type="checkbox" class="sidebar-checkbox" id="sidebar-checkbox" checked>

<!-- Toggleable sidebar -->
<div class="sidebar" id="sidebar">
  <div class="sidebar-item">
    <p>NGS training</p>
  </div>

  <nav class="sidebar-nav">
    <a class="sidebar-nav-item" href="/NGSBASE//">Home</a>

    

    
    
      
        
      
    
      
    
      
        
          <a class="sidebar-nav-item" href="http://localhost:4000/lectures/01.welcome/">Welcome and introduction</a>
        
      
    
      
        
          <a class="sidebar-nav-item" href="http://localhost:4000/lectures/02.setup_tutorial/">Enviroment installation</a>
        
      
    
      
    
      
        
          <a class="sidebar-nav-item" href="http://localhost:4000/lectures/08.fastqc/">Quality Control and Data Preprocessing (I)</a>
        
      
    
      
        
          <a class="sidebar-nav-item active" href="http://localhost:4000/lectures/09.quality_control/">Quality control & Adapter trimming (II)</a>
        
      
    
      
        
          <a class="sidebar-nav-item" href="http://localhost:4000/lectures/10.docker/">Everything in a box. Docker and Galaxy</a>
        
      
    

    <!--<a class="sidebar-nav-item" href="/archive/v2.0.zip">Download</a>-->
    <a class="sidebar-nav-item" href="https://bioinfo-dirty-jobs.github.io/NGSBASE">GitHub project</a>
    <span class="sidebar-nav-item">Currently v2.0</span>
  </nav>

  <div class="sidebar-item">
    <p>
      &copy; 2018. All rights reserved.
    </p>
  </div>
</div>


    <!-- Wrap is the content to shift when toggling the sidebar. We wrap the
         content to avoid any CSS collisions with our real content. -->
    <div class="wrap">
      <div class="masthead">
        <div class="container">
          <h3 class="masthead-title">
            <a href="http://localhost:4000/" title="Home">Training Course on Next generation sequencing</a>
            <!--<p></p><small>Quality control & Adapter trimming (II)</small></p>-->
          </h3>
        </div>
      </div>

      <div class="container content">
        <div class="page">
  <h1 class="page-title">Quality control & Adapter trimming (II)</h1>
  <p class="note">Quality control & Adapter trimming.</p>
  
    <!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})

});
</script>

<div id="toc"></div>

  
  <!-- Common URLs: Tools -->

<!-- Common URLs: File Formats -->

<!-- External URLs -->

<h1 id="preliminaries">Preliminaries</h1>
<p>================================================================================</p>

<h1 id="software-used-in-this-practical">Software used in this practical:</h1>
<hr />

<ul>
  <li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc" title="FastQC home page">FastQC</a> : A quality control tool for high-throughput sequence data.</li>
  <li><a href="http://code.google.com/p/cutadapt" title="cutadapt home page">cutadapt</a> : A tool to remove adapter sequences from high-throughput sequencing data.</li>
</ul>

<h2 id="file-formats-explored-in-this-practical">File formats explored in this practical:</h2>

<p><strong>FastQ</strong>. See:</p>

<ul>
  <li><a href="http://en.wikipedia.org/wiki/FASTQ_format" title="FastQC in Wikipedia">Wikipedia</a>.</li>
  <li><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2847217" title="FastQC original NAR publication">NAR 2010</a>.</li>
</ul>

<h2 id="data-used-in-this-practical">Data used in this practical:</h2>

<ul>
  <li><strong>f010_raw_mirna.fastq</strong>: RNA-Seq of a <a href="http://en.wikipedia.org/wiki/MicroRNA">microRNA</a> sample.</li>
</ul>

<h1 id="overview">Overview</h1>
<p>================================================================================</p>

<ol>
  <li>Use <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc" title="FastQC home page">FastQC</a> to explore the raw data.</li>
  <li>Use <a href="http://code.google.com/p/cutadapt" title="cutadapt home page">cutadapt</a> to remove adapters.</li>
  <li>Use <a href="http://code.google.com/p/cutadapt" title="cutadapt home page">cutadapt</a> to filter reads based on quality.</li>
  <li>Use <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc" title="FastQC home page">FastQC</a> to explore the filtered data.</li>
</ol>

<h1 id="exercise">Exercise</h1>
<p>================================================================================</p>

<p>Create an empty directory to work in the exercise and copy or download the raw data to it:</p>
<div class="highlighter-rouge"><pre class="highlight"><code>    mkdir -p ~/Desktop/quality_control_data/
     cd ~/Desktop/quality_control_data/

&lt;!-- new and clean data directory in the sandbox
    git clone https://github.com/bioinfo-dirty-jobs/Data.git
    cd Data

	cp -r quality_control/ ~/Desktop/quality_control/
    cd ~/Desktop/quality_control_data/
--&gt;
</code></pre>
</div>

<h1 id="explore-the-raw-data-using-some-linux-shell-commands">Explore the raw data using some Linux shell commands</h1>
<hr />

<p>The file <strong>f010_raw_mirna.fastq.gz</strong> contains reads form a microRNA sequencing experiment.
Use the command <code class="highlighter-rouge">head</code> to have a view of the first lines of the file:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>zcat f010_raw_mirna.fastq.gz|head
gunzip f010_raw_mirna.fastq.gz
</code></pre>
</div>

<p>Use the command <code class="highlighter-rouge">wc</code> to count how many reads are there in the file (remember you have to divide by 4)</p>

<div class="highlighter-rouge"><pre class="highlight"><code>wc -l f010_raw_mirna.fastq
</code></pre>
</div>

<h1 id="explore-the-raw-data-quality-using-fastqc">Explore the raw data quality using FastQC</h1>
<hr />

<p>First create a directory to store the results of the fastqc analysis:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>mkdir f020_res_fastqc
</code></pre>
</div>

<p>Then execute <code class="highlighter-rouge">fastqc</code> storing the results in the created directory (option <code class="highlighter-rouge">-o</code>):</p>

<div class="highlighter-rouge"><pre class="highlight"><code>fastqc -o f020_res_fastqc f010_raw_mirna.fastq
</code></pre>
</div>

<p>Find the results in the <strong>fastqc_report.html</strong> file and discus them.</p>

<p>There are many <em>Overrepresented sequences</em>. 
Explore whether some of them correspond to miRNAs using the <a href="http://www.mirbase.org/search.shtml">miRBase search</a> <strong>By sequence</strong> utility.</p>

<h1 id="handling-adapters">Handling adapters</h1>
<hr />

<p>There are 2 known adapters used in this experiment:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA
CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC
</code></pre>
</div>

<p>Use the command grep to see whether they are still present in your data:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>grep "CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA" f010_raw_mirna.fastq 
grep "CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC" f010_raw_mirna.fastq 
</code></pre>
</div>

<p>Do the sequences appear systematically at the beginning or at the end of the reads?</p>

<p>But the adapters could also appear in the <em>reverse</em>, <em>complementary</em> or <em>reverse complementary</em> mode.</p>

<p>Compute the <em>reverse</em>, <em>complementary</em> and the <em>reverse complementary</em> sequences of the two adapters,
and find out which of them appear in your data.</p>

<p>To compute those sequences you can use some online resources as the one in:
<a href="http://www.bioinformatics.org/sms/rev_comp.html">http://www.bioinformatics.org/sms/rev_comp.html</a></p>

<div class="highlighter-rouge"><pre class="highlight"><code>&lt;!--
Or to use R-Bioconductor to compute their reverse, complementary and reverse complementary.

library (Biostrings)
myseq &lt;- DNAString ("CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC")
reverse (myseq)
complement (myseq)
reverseComplement (myseq)

--&gt;
</code></pre>
</div>
<p>Use grep form Linux shell to find out which of the versions of the adapter is in your data.</p>

<h3 id="adapter-1">Adapter 1</h3>

<div class="highlighter-rouge"><pre class="highlight"><code>grep CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA f010_raw_mirna.fastq | wc -l  ## present in the sample (at the beginning of the reads)

grep GACCCTTTAGTGGTATTTGCACTTTACAGAAACCTAAACCCTTAGAATATTCAAGACATACTCTGGTGAGATTTTT f010_raw_mirna.fastq | wc -l 

grep TTTTTAGAGTGGTCTCATACAGAACTTATAAGATTCCCAAATCCAAAGACATTTCACGTTTATGGTGATTTCCCAG f010_raw_mirna.fastq | wc -l  ## present in the sample (at the end of the read) ... but not so numerous

grep AAAAATCTCACCAGAGTATGTCTTGAATATTCTAAGGGTTTAGGTTTCTGTAAAGTGCAAATACCACTAAAGGGTC f010_raw_mirna.fastq | wc -l 
</code></pre>
</div>

<p>But sometimes the adapter does not appear complete. 
It may be there just the first part:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>grep CTGGGAAATCACCATAAACGTGAAATGTCTTTGGA f010_raw_mirna.fastq | wc -l 

grep GACCCTTTAGTGGTATTTGCACTTTACAGAAACCT f010_raw_mirna.fastq | wc -l 

grep TTTTTAGAGTGGTCTCATACAGAACTTATAAGATT f010_raw_mirna.fastq | wc -l 

grep AAAAATCTCACCAGAGTATGTCTTGAATATTCTAA f010_raw_mirna.fastq | wc -l 
</code></pre>
</div>

<p>or the end part of it:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>grep AATCTTATAAGTTCTGTATGAGACCACTCTAAAAA f010_raw_mirna.fastq | wc -l 

grep TTAGAATATTCAAGACATACTCTGGTGAGATTTTT f010_raw_mirna.fastq | wc -l 

grep TCCAAAGACATTTCACGTTTATGGTGATTTCCCAG f010_raw_mirna.fastq | wc -l 

grep AGGTTTCTGTAAAGTGCAAATACCACTAAAGGGTC f010_raw_mirna.fastq | wc -l 
</code></pre>
</div>

<p>NOTE: in the code above I did cut just the 35 first or last nucleotides of the primer in its different arrangements, 
but this is an arbitrary length. 
We are just trying to discover which of the arrangements are present in our sample
and whether there are allocated in the 5’ or in the 6’ end.</p>

<h3 id="adapter-2">Adapter 2</h3>

<div class="highlighter-rouge"><pre class="highlight"><code>grep CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC f010_raw_mirna.fastq | wc -l   ## present in the sample (at the end of the read) ... but not so numerous

grep GAAAAAAAGCAGGAAAGGTGTTCTATATATTTCGGTTCTTTAGCTTTATGAAAGTTCAATGCCATTCG f010_raw_mirna.fastq | wc -l 

grep GCTTACCGTAACTTGAAAGTATTTCGATTTCTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG f010_raw_mirna.fastq | wc -l   ## present in the sample (at the beginning of the reads)

grep CGAATGGCATTGAACTTTCATAAAGCTAAAGAACCGAAATATATAGAACACCTTTCCTGCTTTTTTTC f010_raw_mirna.fastq | wc -l 
</code></pre>
</div>

<p>As before, sometimes the adapter does not appear complete. 
It may be there just the first part:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>grep CTTTTTTTCGTCCTTTCCACAAGATATATA f010_raw_mirna.fastq | wc -l 

grep GAAAAAAAGCAGGAAAGGTGTTCTATATAT f010_raw_mirna.fastq | wc -l 

grep GCTTACCGTAACTTGAAAGTATTTCGATTT f010_raw_mirna.fastq | wc -l 

grep CGAATGGCATTGAACTTTCATAAAGCTAAA f010_raw_mirna.fastq | wc -l 
</code></pre>
</div>

<p>or the end part of it:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>grep AAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC f010_raw_mirna.fastq | wc -l 

grep TTCGGTTCTTTAGCTTTATGAAAGTTCAATGCCATTCG f010_raw_mirna.fastq | wc -l 

grep CTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG f010_raw_mirna.fastq | wc -l 

grep GAACCGAAATATATAGAACACCTTTCCTGCTTTTTTTC f010_raw_mirna.fastq | wc -l 
</code></pre>
</div>

<h2 id="use-cutadapt-to-make-an-adapter-trimming-of-the-reads">Use cutadapt to make an adapter trimming of the reads.</h2>

<p>Check the options:</p>

<ul>
  <li><strong>-a</strong> for adapter to the 3’ end.</li>
  <li><strong>-g</strong> for adapter to the 5’ end.</li>
</ul>

<p>you can find the help of the program typing <code class="highlighter-rouge">cutadapt -h</code> in the shell.</p>

<p>To get read of the the adapters found in our data we run <a href="http://code.google.com/p/cutadapt" title="cutadapt home page">cutadapt</a> several times:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>cutadapt -g CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA -o f030_mirna_trim1.fastq f010_raw_mirna.fastq

cutadapt -a TTTTTAGAGTGGTCTCATACAGAACTTATAAGATTCCCAAATCCAAAGACATTTCACGTTTATGGTGATTTCCCAG -o f030_mirna_trim2.fastq f030_mirna_trim1.fastq

cutadapt -g GCTTACCGTAACTTGAAAGTATTTCGATTTCTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG         -o f030_mirna_trim3.fastq f030_mirna_trim2.fastq

cutadapt -a CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC         -o f030_mirna_trim4.fastq f030_mirna_trim3.fastq
</code></pre>
</div>

<p>Now you can <code class="highlighter-rouge">grep</code> again searching for the adapters</p>

<h3 id="adapter-1-1">Adapter 1</h3>

<div class="highlighter-rouge"><pre class="highlight"><code>grep CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA f030_mirna_trim4.fastq | wc -l

#grep GACCCTTTAGTGGTATTTGCACTTTACAGAAACCTAAACCCTTAGAATATTCAAGACATACTCTGGTGAGATTTTT f030_mirna_trim4.fastq | wc -l 

grep TTTTTAGAGTGGTCTCATACAGAACTTATAAGATTCCCAAATCCAAAGACATTTCACGTTTATGGTGATTTCCCAG f030_mirna_trim4.fastq | wc -l

#grep AAAAATCTCACCAGAGTATGTCTTGAATATTCTAAGGGTTTAGGTTTCTGTAAAGTGCAAATACCACTAAAGGGTC f030_mirna_trim4.fastq | wc -l 
</code></pre>
</div>

<h3 id="adapter-2-1">Adapter 2</h3>

<div class="highlighter-rouge"><pre class="highlight"><code>grep CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC f030_mirna_trim4.fastq | wc -l

#grep GAAAAAAAGCAGGAAAGGTGTTCTATATATTTCGGTTCTTTAGCTTTATGAAAGTTCAATGCCATTCG f030_mirna_trim4.fastq | wc -l 

grep GCTTACCGTAACTTGAAAGTATTTCGATTTCTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG f030_mirna_trim4.fastq | wc -l

#grep CGAATGGCATTGAACTTTCATAAAGCTAAAGAACCGAAATATATAGAACACCTTTCCTGCTTTTTTTC f030_mirna_trim4.fastq | wc -l 
</code></pre>
</div>

<h2 id="explore-the-quality-of-the-trimmed-file-using-fastqc">Explore the quality of the trimmed file using FastQC</h2>

<p>Check the data quality again using fastqc:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>mkdir f040_res_fastqc_trimmed
fastqc -o f040_res_fastqc_trimmed f030_mirna_trim4.fastq
</code></pre>
</div>

<p>Some of the reads seems to be too short and some others may not have enough quality.</p>

<h2 id="use-cutadapt-to-filter-reads-by-quality-and-length">Use cutadapt to filter reads by quality and length.</h2>

<p>Check the options:</p>

<ul>
  <li><strong>-q</strong> quality cutoff.</li>
  <li><strong>-m</strong> minimum length.</li>
  <li><strong>-M</strong> maximum length.</li>
</ul>

<p>you can find the help of the program typing <code class="highlighter-rouge">cutadapt -h</code> in the shell.</p>

<p>Run cutadapt for length and quality purge of the reads.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>cutadapt -m 17 -M 30 -q 10 -o f040_mirna_cut.fastq f030_mirna_trim4.fastq
</code></pre>
</div>

<p>Check the data quality again using fastqc:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>mkdir f050_res_fastqc_trimmed_purged

fastqc -o f050_res_fastqc_trimmed_purged f040_mirna_cut.fastq
</code></pre>
</div>

<p>Explore again the <em>Overrepresented sequences</em> in <a href="http://www.mirbase.org/" title="miRBase: a searchable database of published miRNA">mirbase</a> (Search -&gt; By sequence).</p>

<p>Count how many reads are left for the analysis (divide by 4)</p>

<div class="highlighter-rouge"><pre class="highlight"><code>wc -l f010_raw_mirna.fastq

wc -l f040_mirna_cut.fastq # OPTIONAL
fastx_collapse f010_raw_mirna.fastq &gt;f010_raw_mirna.fastq.collpse
</code></pre>
</div>


</div>

      </div>
    </div>

    <label for="sidebar-checkbox" class="sidebar-toggle"></label>

    <script>
      (function(document) {
        var toggle = document.querySelector('.sidebar-toggle');
        var sidebar = document.querySelector('#sidebar');
        var checkbox = document.querySelector('#sidebar-checkbox');

        document.addEventListener('click', function(e) {
          var target = e.target;

          if(!checkbox.checked ||
             sidebar.contains(target) ||
             (target === checkbox || target === toggle)) return;

          checkbox.checked = false;
        }, false);
      })(document);
    </script>
  <!--<footer>
        <p>This project is maintained by  Maurizio Polano<br/></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>

-->
  </body>
</html>
