---
layout: page
title: Welcome and introduction
summary: "Agenda, resources and other useful information"
---

## Welcome!

Welcome to this introductory course about _NGS data analysis_. During this course you will learn the basics of NGS data analysis in a Linux environment, current used software and best practices will be explained.
This course is focused in NGS data alignment for both DNA and RNA (RNA-seq), variant calling, differential expression analysis and data visualization.
This is course is scheduled for a 2 days and assumes a very basic knowledge of NGS data analysis and Linux.
All materials in this is course are free and open and derived from some pulic avaible online course.

[DropBox folder](https://www.dropbox.com/sh/4qkqch7gyt888h7/AABD_i9ShwryfAqGeJ0yqqF3a)

[data](https://drive.google.com/open?id=1ywFJ9OyKn0Bo9AhxX2nUuKC3Fr4wLdL6)
[TUTORIAL RNA](https://www.dropbox.com/s/ewlkswl9nd31xz9/RNA_all.tar.bzip2?dl=0)
[data2]()

Here you can found and Interesting video that explain NGS sequincing

[![Sequencing](https://www.youtube.com/watch?v=MvuYATh7Y74/0.jpg)](https://www.youtube.com/watch?v=MvuYATh7Y74)
---
### Learning outcomes

After this course you should be able to:

- Start to move on linux system 
- Know the precincipal formats  use on  NGS 
- Evaluate quality and clean fastq 
- Map and visualize some alignment
- Know some base of analisys



and you will be exposed to these software:


- [FASTC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
- [SAMTOOLS](http://www.htslib.org/doc/samtools.html)
- [bedtools](http://bedtools.readthedocs.io/en/latest/)
- [bwa](http://bio-bwa.sourceforge.net/)
- [bowtie2](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml)
- [multiQC](http://multiqc.info/)
- [STAR](https://github.com/alexdobin/STAR)
- [vt](https://genome.sph.umich.edu/wiki/Vt)
- [freebays](https://github.com/ekg/freebayes)
- [GATK](https://software.broadinstitute.org/gatk/)
- [NEXTFLOW](https://www.nextflow.io/)
- [KALLISTO](https://pachterlab.github.io/kallisto/about)
- [SLEUTH](https://github.com/pachterlab/sleuth)
- [picards](https://broadinstitute.github.io/picard/)

NGS data can be big, very big, huge! Biology is now a Big Data science
There are not many web-based or graphical applications to perform analysis yet, sorry.
Most tools developed to work on Linux, many command line programs
How to work in NGS?
- Small datasets (<1TB): workstations
- Medium sized datasets (<100TB): clusters
- Big datasets (100TB-20PB): big clusters and/or cloud based solutions

Exercises during this course the NGS alignment will be done using the human chromosome 21 or MT as a reference genome. By doing this we can speed up exercises and avoid using too much memory. Under real circumstances, when using the standard reference genome, all the commands are exactly the same
Software has been already installed to save time, so you are not expected to download and install all the software it is going to be used. However, it's usually good to learn the basics of software installation in Linux, there is an optional session at the end of the first day for those that want to learn how to install NGS software in a standard Linux. We use conda for install most of the tools.

<img src="{{site.url}}/images/p3.jpeg" width="600">


---
### We assume you are familiar with:

- Next generation sequencing in general
- Be patient
- Please see this video [How to sequence the human genome - Mark J. Kiel](https://www.youtube.com/watch?v=MvuYATh7Y74)

<img src="{{site.url}}/images/patience.jpg" width="600" >





---
# Agenda

**12 march 2018**

Title | Lecture/Practical
------|-------------------
**Introduction & overview of NGS (RNA and DNA sequencing)** | Lecture
**Introduction to Linux and R and command line interface** | Lecture / Practical
**NGS data format** | Lecture / Practical

**13 march 2018**

Title | Lecture/Practical
------|-------------------
**Bioinformatics approach (galaxy nextflow,and Ad hoc scripting)** | Lecture /Practical
**RNASeq Analysis Workflow and planning (Quality Control, Alignment, Gene expression quantification, Differential expression analysis)**| Lecture /Practical
**Exome and target sequencing workflow (alignment, SNP calling)** | Practical
**ChIPSeq workflow** | Lecture





