---
layout: page
title: Welcome and introduction
summary: "Agenda, resources and other useful information"
---

## Welcome!

In this part of the course, we will present the main steps that are commonly used to process and to analyze cancer sequencing data for detect fusion transcript genes. We will focus on RNA-seq data and we will provide command lines that allow detecting fusions transcript genes using some of the most used programs.

<img src="{{site.url}}/images/miniom.jpeg" width="600">




---
### Learning outcomes

After this course you should be able to:

- install  an use  software for detect fusion transcript
- understand the main process for detect fusion genes and detect some false positive 
- choose the right tools for the data you have


and you will be exposed to these software:


- **STAR-FUSION** [https://github.com/STAR-Fusion/STAR-Fusion/wiki]
- **JAFFA** [https://github.com/Oshlack/JAFFA/wiki]
- Fusioncather [https://github.com/ndaniel/fusioncatcher]
- PRADA [http://bioinformatics.mdanderson.org/main/PRADA:Overview]
- Defuse [https://github.com/FusionInspector/FusionInspector]

Jaffa can be tested in our enviromnt with 16G RAM. results (STAR-FUSION we can se the results usin my cluster).

<img src="{{site.url}}/images/p3.jpeg" width="600">


---
### We assume you are familiar with:

-  Next generation sequencing in general
-  RNA-seq
- bash



---
# Agenda

**September 7*

Title | Lecture/Practical
------|-------------------
**Basic information** | Practical
**JAFFA** | Introduction
**JAFFA** | Practical
**Star-fusion** | Lecture/Pratical
**Exploration data from other softwares like defuse** | Practical
