---
layout: page
title:  Quality Control and Data Preprocessing (I)
summary: "Quality Control and Data Preprocessing."
---


# FastQ Format
================================================================================

- Standard Format for NGS data
- Conversion can be done from _sff_, _fasta + qual_, ... 
- Extension of the Fasta format
- Text-based formats (easy to use!)
- If not compressed, it can be huge


<http://en.wikipedia.org/wiki/FASTQ_format>


# Quality measurements
================================================================================

Base-calling __error probabilities__ are reported by sequencers.

Usually in __Phred__ (quality) score.

Usually coded by ASCII characters


Phred score
------------

$$Q = -10 log_{10} P$$

$$P = 10^{\frac{-Q}{10}}$$



[Pred](http://en.wikipedia.org/wiki/Phred_quality_score#Definition)




# NGS Data Preprocessing Steps
---


- File parsing: convert to __fastq__ format form __sff__, __fasta__ + __qual__ ...
- Split [multiplex](http://www.illumina.com/technology/multiplexing_sequencing_assay.ilmn "Multiplex Sequencing Assay") samples.

- Quality Control of the raw data.

- Filtering and trimming reads by quality.
- Adapter trimming

- Quality Control of the trimmed and filtered reads



# Software
---


- __FastQC__:
    - quality control
    - some filtering ...

[fastqC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc)



- __Cutadapt__: 
    - adapter trimming 
	- filter reads by length (short, long)
	- filter reads by quality

[cutadpt] (<http://code.google.com/p/cutadapt>)




<!-- FastQC Images -->

Per Base Sequence Quality
================================================================================

<img src="{{site.url}}/images/per_base_quality.png">


Per Sequence Quality
================================================================================

<img src="{{site.url}}/images/per_sequence_quality.png">
Per Base Sequence Content
================================================================================

<img src="{{site.url}}/images/per_base_sequence_content.png">
Per Base GC Content
================================================================================

<img src="{{site.url}}/images/per_base_gc_content.png">
Per Sequence Nucleotide Content
================================================================================

<img src="{{site.url}}/images/per_sequence_gc_content.png">
Per Base N Content
===============================================================================

<img src="{{site.url}}/images/per_base_n_content.png">
Sequence Length Distribution
===============================================================================

<img src="{{site.url}}/images/sequence_length_distribution.png">
Duplicate Sequences Distribution
================================================================================

<img src="{{site.url}}/images/duplication_levels.png">
Over-represented Kmers
================================================================================

<img src="{{site.url}}/images/kmer_profiles.png">


More FastQ examples and documentation
================================================================================

... may be found at [FastQ home page](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)

- Example Reports



Sequencing process: PCR primers 
================================================================================
<img src="{{site.url}}/images/pcr_adaptors.png">


Sequencing process: PCR primers 
================================================================================
<img src="{{site.url}}/images/illumina_cluster_generation.png">


NGS adaptors and Cutadapt
================================================================================

<img src="{{site.url}}/images/adaptors.png">


## MULTIQC

Aggregate results from bioinformatics analyses across many samples into a single report. MultiQC searches a given directory for analysis logs and compiles a HTML report. It's a general use tool, perfect for summarising the output from numerous bioinformatics tools [MULTIQC](http://multiqc.info/)
<img src="{{site.url}}/images/multiqc.png">